File: src/main/java/com/github/sorusclient/client/adapter/Vertex.java
/*
 * Copyright (c) 2022. Danterus
 * Copyright (c) 2022. Sorus Contributors
 *
 * SPDX-License-Identifier: MPL-2.0
 */

package com.github.sorusclient.client.adapter;

import com.github.sorusclient.client.util.Color;

public class Vertex {

    private Point point;
    private Color color;

    public Point getPoint() {
        return point;
    }

    public Color getColor() {
        return color;
    }

    public Vertex setPoint(Point point) {
        this.point = point;
        return this;
    }

    public Vertex setColor(Color color) {
        this.color = color;
        return this;
    }

}

File: src/main/kotlin/com/github/sorusclient/client/adapter/Vertex.kt
/*
 * Copyright (c) 2022. Danterus
 * Copyright (c) 2022. Sorus Contributors
 *
 * SPDX-License-Identifier: MPL-2.0
 */

package com.github.sorusclient.client.adapter

import com.github.sorusclient.client.util.Color

class Vertex {
    lateinit var point: Point
        private set
    lateinit var color: Color
        private set

    fun setPoint(point: Point): Vertex {
        this.point = point
        return this
    }

    fun setColor(color: Color): Vertex {
        this.color = color
        return this
    }
}